mocks/firestore.js

```diff
@@ -464,12 +467,14 @@ FakeFirestore.CollectionReference = class extends FakeFirestore.Query {
 
   _get() {
     // Make sure we have a 'good enough' document reference
-    const records = this._records();
-    records.forEach(rec => {
-      rec._ref = new FakeFirestore.DocumentReference(rec.id, this, this.firestore);
-    });
+    const records = this._records().map(rec => ({
+      ...rec,
+      _ref: new FakeFirestore.DocumentReference(rec.id, this, this.firestore),
+    }));
+    // Firestore does not return documents with no local data
+    const onlyLocal = records.filter(r => Object.keys(r._ref._get().data()).length > 0);
```

Comment:
Can we actually get this in a separate PR? It feels unrelated to the `listDocuments` change and in case this _does_ introduce something that is breaking, we can roll it back without losing the `listDocument` changes

Reply:
a year or so later - it has been removed